src/coreComponents/constitutive/capillaryPressure/JFunctionCapillaryPressure.cpp

```diff
@@ -250,6 +250,7 @@ void JFunctionCapillaryPressure::saveConvergedRockState( arrayView2d< real64 con
     {
       permeability = convergedPermeability[ei][0][2];
     }
+    GEOS_ERROR_IF( !isZero( permeability ), GEOS_FMT( "{}: zero permeability in J-function capillary pressure", getFullName() ));
```

Comment:
```suggestion
    GEOS_ERROR_IF( permeability < LvArray::NumericLimits<real64>::epsilon, GEOS_FMT( "{}: zero permeability in J-function capillary pressure", getFullName() ));
```
I would completely ban negative values as well.